### PR TITLE
`rhel86` various fixes for the `azure-rhui` image

### PR DESCRIPTION
**Pull request by gicmo**
- Sync the package set with the existing market place image
- Update sysconfig to set `kernel-core` as default kernel
- Remove `authconfig` call
- Import GPG keys from the tree(!) via a new `org.osbuild.rpm` stage option.

NB: the last point needs https://github.com/osbuild/osbuild/pull/985

### CODE REVIEW

File: internal/distro/image_config.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ type ImageConfig struct {
 	DefaultTarget       string
 	Sysconfig           []*osbuild2.SysconfigStageOptions
 
+	GPGKeyFiles []string
```
Comment (thozza) [inline internal/distro/image_config.go:23]: I would suggest to add "RPM" to the variable name to make it obvious what type of GPG keys these are.

In addition, you need to extend the `InheritFrom` method due to this new member.

**Review by thozza — CHANGES_REQUESTED** [review]
The `InheritFrom` method needs to be extended, otherwise this looks good to me.

File: internal/distro/image_config.go (hunk 1 above)
Comment (gicmo) [reply internal/distro/image_config.go:23]: Good catch! We should enhance that to use reflection in a future version (and maybe switch it do clone and override ;) ;)

Comment (thozza) [reply internal/distro/image_config.go:23]: _nods_

**Comment by thozza** [comment]
@gicmo one more note, I think that the osbuild version dependency in the SPEC should be bumped with this PR, because it depends on a specific osbuild version (the one next one - not released yet)

**Comment by gicmo** [comment]
@thozza I did not do that because I was wondering if a downstream patch for osbuild is enough of if we can deploy `main`. I am shaky on those details.

**Comment by thozza** [comment]
> @thozza I did not do that because I was wondering if a downstream patch for osbuild is enough of if we can deploy `main`. I am shaky on those details.

If we don't do it now, we will IMO most probably forget to do it at all. If we would deploy from `main`, we would use a CI build and that already uses the bumped version (https://gitlab.com/osbuild/ci/osbuild/-/jobs/2162857403#L1028) so this is not a blocker.

**Comment by gicmo** [comment]
> > @thozza I did not do that because I was wondering if a downstream patch for osbuild is enough of if we can deploy `main`. I am shaky on those details.
> 
> Yeah, this is a bit fuzzy...
> 
> If we don't do it now, we will IMO most probably forget to do it at all. If we would deploy from `main`, we would use a CI build and that already uses the bumped version (https://gitlab.com/osbuild/ci/osbuild/-/jobs/2162857403#L1028) so this is not a blocker.

Excellent point! Will fix as well!

File: internal/distro/image_config.go (hunk 1 above)
Comment (gicmo) [reply internal/distro/image_config.go:24]: I added a comment for now, can always rename it later if it collides with something.

**Review by thozza — APPROVED** [review]
LGTM

File: internal/distro/image_config.go (hunk 1 above)
Comment (thozza) [reply internal/distro/image_config.go:24]: It probably won't collide, I would just find "RPM" in the variable name easier to understand when one uses the structure, compared to a comment in its definition. But I agree with you and think that the current state is OK to merge.

**Review by thozza — DISMISSED** [review]

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**